### PR TITLE
OData implementation

### PR DESCRIPTION
**Pull request by alaraalara**
1. Added a get method in OData.cs that handles both login and data retrievals. 
2. Added a ODataExample.cs file to demonstrate the login and data retrievals in both OData version 3 and 4.
3. Console Application solution uses dll files generated by RestClient as a reference, instead of Nuget packages.

### CODE REVIEW

File: Acumatica.RESTClient/BaseApi/ODataAPI.cs [hunk 1]
```diff
@@ -1,23 +1,58 @@
...
-
 using Acumatica.RESTClient.Client;
-
 using RestSharp;
+using RestSharp.Authenticators;
 
 namespace Acumatica.RESTClient.Api
 {
+   
     public class ODataAPI : BaseApi
```
Comment (Dmitrii-Naumov) [inline Acumatica.RESTClient/BaseApi/ODataAPI.cs:10]: If we use "RestRequest" for sending the API call, does it make sense to have it derived from BaseApi?

File: Acumatica.RESTClient/BaseApi/ODataAPI.cs [hunk 2]
```diff
@@ -1,23 +1,58 @@
...
+            basePath = configuration.BasePath;
         }
+        /*
         public string Get(string path, string select = null, string filter = null, string expand = null, string custom = null, int? skip = null, int? top = null)
         {
             ApiResponse<string> localVarResponse = GetWithHttpInfo(path, select, filter, expand, custom, skip, top);
             return localVarResponse.Data;
         }
+        */
+        public string Get(string version, string resource = null, Dictionary<string, string> parameters = null, string username = null, string password = null)
```
Comment (Dmitrii-Naumov) [inline Acumatica.RESTClient/BaseApi/ODataAPI.cs:26]: Ideally the OData call must work with both OAuth authentication and base authentication. In the this implementation only base is possible, right?

File: Acumatica.RESTClient/BaseApi/ODataAPI.cs [hunk 3]
```diff
@@ -1,23 +1,58 @@
...
+        {
+            if(username != null && password != null)
+            {
+                //get for login
+                basePath += "/" + version;
```
Comment (Dmitrii-Naumov) [inline Acumatica.RESTClient/BaseApi/ODataAPI.cs:31]: I don't like the version being string parameter. It leaves a lot of room for a human error. It'd be better that only correct values can be passed to the method

File: Acumatica.RESTClient/BaseApi/ODataAPI.cs [hunk 4]
```diff
@@ -1,23 +1,58 @@
...
+                return apiresponse.Data;
+            }
+            else
+            {
+                var request = new RestRequest(resource);
+                if(parameters != null)
+                {
+                    foreach (var parameter in parameters)
+                    {
+                        request.AddParameter(parameter.Key, parameter.Value);
```
Comment (Dmitrii-Naumov) [inline Acumatica.RESTClient/BaseApi/ODataAPI.cs:46]: As for parameters, it looks like in this implementation a developer needs to pass proper strings as parameters. 
However, it'd be better if the method signature makes it clear what parameters are available and eliminates a chance of passing incorrect parameters

File: Acumatica.RESTClient/BaseApi/ODataAPI.cs (hunk 3 above)
Comment (Dmitrii-Naumov) [inline Acumatica.RESTClient/BaseApi/ODataAPI.cs:31]: Having basePath as global variable but modifying it in the method body is a bad pattern. If you call this method twice, the basePath becomes invalid

File: Acumatica.RESTClient/BaseApi/ODataAPI.cs [hunk 5]
```diff
@@ -1,23 +1,58 @@
...
             return localVarResponse.Data;
         }
+        */
+        public string Get(string version, string resource = null, Dictionary<string, string> parameters = null, string username = null, string password = null)
+        {
+            if(username != null && password != null)
+            {
+                //get for login
+                basePath += "/" + version;
+                client = new RestClient(basePath);
```
Comment (Dmitrii-Naumov) [inline Acumatica.RESTClient/BaseApi/ODataAPI.cs:32]: For Odata request with multitenant environment the url is not correct here. We need companyID part in the url

File: Acumatica REST API Console Application/ODataExample.cs [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+			var configuration = new Configuration(basePath);
+
+			Console.WriteLine("Testing sign in");  
+			ODataAPI odata = new ODataAPI(configuration);
+			var response = odata.Get(version, null, null, username, password);
+			Console.WriteLine(response);
+
+			Console.WriteLine("Example retrieval of customers with OData Version 3");
+			var parameter = new Dictionary<string, string>();
+			parameter.Add("format", "json");
```
Comment (Dmitrii-Naumov) [inline Acumatica REST API Console Application/ODataExample.cs:23]: I think we should always just use json for Odata, no need to pass it from where you invoke the method

**Review by Dmitrii-Naumov — CHANGES_REQUESTED** [review]
see comments in the code

**Review by alaraalara — COMMENTED** [review]
1. added compatibility with Oauth. 
2. Parameters are now shown in method signature. 
3. Version is converted to an enum variable. 
4. Configurepath method is added for allowing multitenant environment. 
5. Json format is default for Get method.

File: Acumatica.RESTClient/BaseApi/ODataAPI.cs [hunk 7]
```diff
@@ -1,46 +1,151 @@
...
         }
 
+        #endregion
+
+
     }
+
+
+
+    public enum Version
```
Comment (Dmitrii-Naumov) [inline Acumatica.RESTClient/BaseApi/ODataAPI.cs:146]: Let's rename it to ODataVersion

File: Acumatica REST API Console Application/Program.cs [hunk 8]
```diff
@@ -8,27 +8,40 @@
...
 	{
-		const string SiteURL = "http://localhost/demo/";
+		//const string SiteURL = "http://localhost/demo/";
+		const string SiteURL = "http://localhost/MyStoreInstance";
 		const string Username = "admin";
 		const string Password = "123";
 		const string Tenant = null;
 		const string Branch = null;
 		const string Locale = null;
+		const string tokenUrl = "https://localhost/MyStoreInstance/identity/connect/token";
```
Comment (Dmitrii-Naumov) [inline Acumatica REST API Console Application/Program.cs:25]: It seems like this constant is never used. 
Please remove it

**Review by Dmitrii-Naumov — CHANGES_REQUESTED** [review]

**Pull request merged**